provider_filetree.go

```diff
@@ -58,6 +138,22 @@ func isDir(filename string, f os.FileInfo) bool {
 	return f.IsDir()
 }
 
+func watch(watcher *fsnotify.Watcher, root string) error {
```

Comment:
function name is not very explicit/easy to understand at first read. Something like `watchDirFiles` would be better.